cmd/shortener/main.go

```diff
@@ -35,54 +35,52 @@ func randString(length int) string {
 
 func shorten(w http.ResponseWriter, r *http.Request) {
 
-	if r.Method != http.MethodPost {
-		http.Error(w, "Only POST requests are allowed", http.StatusMethodNotAllowed)
-		return
-	}
 	//читаем тело
 	urlFromRequest, _ := io.ReadAll(r.Body)
+
+	_, err := url.Parse(string(urlFromRequest))
+	if err != nil {
+		http.Error(w, "not url", http.StatusBadRequest)
+		return
+	}
 	//генерим короткий url
 	shortURL := randString(8)
 	// сохраняем в мапе
-	urlsStorage[shortURL] = string(urlFromRequest)
+
+	urlsStorage.Store(shortURL, string(urlFromRequest))
 	//заполняем ответ
-	if cfg.Cfg.ServerAddress == "" {
-		cfg.Cfg.ServerAddress = *a
-	}
-	if cfg.Cfg.BaseURL == "" {
-		cfg.Cfg.BaseURL = *b
-	}
 	body := fmt.Sprintf("%s/%s", cfg.Cfg.BaseURL, shortURL)
 	w.Header().Add("Content-Type", "text/plain")
 	w.Header().Add("Host", cfg.Cfg.ServerAddress)
 	w.WriteHeader(http.StatusCreated)
-	_, err := w.Write([]byte(body))
+	_, err = w.Write([]byte(body))
 	if err != nil {
-		panic(err)
+		log.Fatal(err)
```

Comment:
Ни в коем разе! Это приведет к моментальному падению сервера, по сути, из-за разрыва соединения. Только обычное логирование!
```suggestion
		log.Error(err)
```